lighthouse-core/gather/fetcher.js

```diff
@@ -156,18 +158,16 @@ class Fetcher {
     });
     await this.session.sendCommand('Network.disable');
 
-    if (!networkResponse.resource.success || !networkResponse.resource.stream) {
-      const statusCode = networkResponse.resource.httpStatusCode || '';
-      throw new Error(`Loading network resource failed (${statusCode})`);
-    }
-
-    return networkResponse.resource.stream;
+    return {
+      stream: networkResponse.resource.success ? (networkResponse.resource.stream || null) : null,
+      status: networkResponse.resource.httpStatusCode || null,
+    };
```

Comment:
I used `null` to match the expectations of the `RobotsTxt` artifact. I think `null` makes more sense than `undefined` but it leads to some ugly code like this.

Reply:
I have had 3 different versions of a comment written here in various tabs that I've lost with varying levels of concern, but at this point it just SGTM 😆

Reply:
what are the failure cases? `ERR_BLOCKED_BY_CLIENT`? Is `success` set by `netError` _and_ error status codes, or just `netError`? If it's just `netError`, maybe we should mimic real `fetch()` and throw in that case?

Reply:
> If it's just netError, maybe we should mimic real fetch() and throw in that case?

In one iteration of my comment, I discuss with my lost tab that it seems odd that we would throw an exception rather than a warning on notApplicable. It's not a bug, but rather a known edge case, so the new `null` behavior makes more sense. That does suggest we actually make it a warning in RobotsTxt audit.

I'm fine with maintaining the status quo as well and throwing if others prefer that.

Reply:
Yeah, I was thinking in terms of `fetcher` on its own. The gatherer could then do as it saw fit with the response

Reply:
`success` is set to `false` for error status codes, and there is no stream handle to use if content is served wit an error status code.

Reply:
>  there is no stream handle to use if content is served wit an error status code.

that seems like a CDP bug but probably not one worth pursuing until we have a reason to want to look at response bodies from requests that 404 or whatever :)

I think it still could make sense for fetcher to return
- good statusCode: `Promise.resolve({status: number, content: string})`
- bad statusCode: `Promise.resolve({status: number, content: null})`
- netError: `Promise.reject(new Error(netErrorName))`

(and e.g. the `RobotsTxt` gatherer could catch the error and augment the artifact with a failure reason to add to the audit `explanation` string)

but I don't think this is critical today, so I'm fine with this version as well.